Use ASCIILiteral instead of `const char*` for Supplementable key

cdumez opened the pull request:
#### 67e8a2009c5394a9dd03bc78614a938df568b6c9
<pre>
Use ASCIILiteral instead of `const char*` for Supplementable key
<a href="https://bugs.webkit.org/show_bug.cgi?id=267431">https://bugs.webkit.org/show_bug.cgi?id=267431</a>

Reviewed by Geoffrey Garen.

It makes string lifetime clearer.

* Source/WTF/wtf/text/ASCIILiteral.h:
(WTF::ASCIILiteralPtrHash::hash):
(WTF::ASCIILiteralPtrHash::equal):
* Source/WebCore/Modules/async-clipboard/NavigatorClipboard.cpp:
(WebCore::NavigatorClipboard::supplementName):
* Source/WebCore/Modules/async-clipboard/NavigatorClipboard.h:
* Source/WebCore/Modules/audiosession/NavigatorAudioSession.cpp:
(WebCore::NavigatorAudioSession::supplementName):
* Source/WebCore/Modules/audiosession/NavigatorAudioSession.h:
* Source/WebCore/Modules/beacon/NavigatorBeacon.cpp:
(WebCore::NavigatorBeacon::supplementName):
* Source/WebCore/Modules/beacon/NavigatorBeacon.h:
* Source/WebCore/Modules/cache/WindowOrWorkerGlobalScopeCaches.cpp:
(WebCore::DOMWindowCaches::supplementName):
(WebCore::WorkerGlobalScopeCaches::supplementName):
* Source/WebCore/Modules/contact-picker/NavigatorContacts.cpp:
(WebCore::NavigatorContacts::supplementName):
* Source/WebCore/Modules/contact-picker/NavigatorContacts.h:
* Source/WebCore/Modules/cookie-consent/NavigatorCookieConsent.h:
* Source/WebCore/Modules/credentialmanagement/NavigatorCredentials.cpp:
(WebCore::NavigatorCredentials::supplementName):
* Source/WebCore/Modules/credentialmanagement/NavigatorCredentials.h:
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemController.cpp:
(WebCore::MediaKeySystemController::supplementName):
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemController.h:
* Source/WebCore/Modules/gamepad/NavigatorGamepad.cpp:
(WebCore::NavigatorGamepad::supplementName):
* Source/WebCore/Modules/gamepad/NavigatorGamepad.h:
* Source/WebCore/Modules/geolocation/GeolocationController.cpp:
(WebCore::GeolocationController::supplementName):
* Source/WebCore/Modules/geolocation/GeolocationController.h:
* Source/WebCore/Modules/geolocation/NavigatorGeolocation.cpp:
(WebCore::NavigatorGeolocation::supplementName):
* Source/WebCore/Modules/geolocation/NavigatorGeolocation.h:
* Source/WebCore/Modules/indexeddb/WindowOrWorkerGlobalScopeIndexedDatabase.cpp:
(WebCore::DOMWindowIndexedDatabase::supplementName):
(WebCore::WorkerGlobalScopeIndexedDatabase::supplementName):
* Source/WebCore/Modules/mediacapabilities/NavigatorMediaCapabilities.cpp:
(WebCore::NavigatorMediaCapabilities::supplementName):
* Source/WebCore/Modules/mediacapabilities/NavigatorMediaCapabilities.h:
* Source/WebCore/Modules/mediacapabilities/WorkerNavigatorMediaCapabilities.cpp:
(WebCore::WorkerNavigatorMediaCapabilities::supplementName):
* Source/WebCore/Modules/mediacapabilities/WorkerNavigatorMediaCapabilities.h:
* Source/WebCore/Modules/mediasession/NavigatorMediaSession.cpp:
(WebCore::NavigatorMediaSession::supplementName):
* Source/WebCore/Modules/mediasession/NavigatorMediaSession.h:
* Source/WebCore/Modules/mediastream/NavigatorMediaDevices.cpp:
(WebCore::NavigatorMediaDevices::supplementName):
* Source/WebCore/Modules/mediastream/NavigatorMediaDevices.h:
* Source/WebCore/Modules/mediastream/UserMediaController.cpp:
(WebCore::UserMediaController::supplementName):
* Source/WebCore/Modules/mediastream/UserMediaController.h:
* Source/WebCore/Modules/notifications/NotificationController.cpp:
(WebCore::NotificationController::supplementName):
* Source/WebCore/Modules/notifications/NotificationController.h:
* Source/WebCore/Modules/permissions/NavigatorPermissions.cpp:
(WebCore::NavigatorPermissions::supplementName):
* Source/WebCore/Modules/permissions/NavigatorPermissions.h:
* Source/WebCore/Modules/permissions/WorkerNavigatorPermissions.cpp:
(WebCore::WorkerNavigatorPermissions::supplementName):
* Source/WebCore/Modules/permissions/WorkerNavigatorPermissions.h:
* Source/WebCore/Modules/pictureinpicture/DocumentPictureInPicture.h:
(WebCore::DocumentPictureInPicture::supplementName):
* Source/WebCore/Modules/pictureinpicture/HTMLVideoElementPictureInPicture.h:
(WebCore::HTMLVideoElementPictureInPicture::supplementName):
* Source/WebCore/Modules/push-api/ServiceWorkerRegistrationPushAPI.cpp:
(WebCore::ServiceWorkerRegistrationPushAPI::supplementName):
* Source/WebCore/Modules/push-api/ServiceWorkerRegistrationPushAPI.h:
* Source/WebCore/Modules/screen-wake-lock/NavigatorScreenWakeLock.cpp:
(WebCore::NavigatorScreenWakeLock::supplementName):
* Source/WebCore/Modules/screen-wake-lock/NavigatorScreenWakeLock.h:
* Source/WebCore/Modules/speech/LocalDOMWindowSpeechSynthesis.cpp:
(WebCore::LocalDOMWindowSpeechSynthesis::supplementName):
* Source/WebCore/Modules/speech/LocalDOMWindowSpeechSynthesis.h:
* Source/WebCore/Modules/webdriver/NavigatorWebDriver.cpp:
(WebCore::NavigatorWebDriver::supplementName):
* Source/WebCore/Modules/webdriver/NavigatorWebDriver.h:
* Source/WebCore/css/DOMCSSPaintWorklet.cpp:
(WebCore::DOMCSSPaintWorklet::supplementName):
* Source/WebCore/css/DOMCSSPaintWorklet.h:
* Source/WebCore/css/DOMCSSRegisterCustomProperty.cpp:
(WebCore::DOMCSSRegisterCustomProperty::supplementName):
* Source/WebCore/css/DOMCSSRegisterCustomProperty.h:
* Source/WebCore/css/typedom/CSSNumericFactory.cpp:
(WebCore::CSSNumericFactory::supplementName):
* Source/WebCore/css/typedom/CSSNumericFactory.h:
* Source/WebCore/dom/DeviceMotionController.cpp:
(WebCore::DeviceMotionController::supplementName):
* Source/WebCore/dom/DeviceMotionController.h:
* Source/WebCore/dom/DeviceOrientationController.cpp:
(WebCore::DeviceOrientationController::supplementName):
* Source/WebCore/dom/DeviceOrientationController.h:
* Source/WebCore/dom/DocumentStorageAccess.cpp:
(WebCore::DocumentStorageAccess::supplementName):
* Source/WebCore/dom/DocumentStorageAccess.h:
* Source/WebCore/html/NavigatorUserActivation.cpp:
(WebCore::NavigatorUserActivation::supplementName):
* Source/WebCore/html/NavigatorUserActivation.h:
* Source/WebCore/page/NavigatorIsLoggedIn.cpp:
(WebCore::NavigatorIsLoggedIn::supplementName):
* Source/WebCore/page/NavigatorIsLoggedIn.h:
* Source/WebCore/platform/Supplementable.h:
(WebCore::Supplement::provideTo):
(WebCore::Supplement::from):
(WebCore::Supplementable::provideSupplement):
(WebCore::Supplementable::removeSupplement):
(WebCore::Supplementable::requireSupplement):
* Source/WebCore/testing/InternalSettings.cpp:
(WebCore::InternalSettings::supplementName):
* Source/WebCore/testing/InternalSettings.h:
* Source/WebCore/workers/service/background-fetch/ServiceWorkerRegistrationBackgroundFetchAPI.cpp:
(WebCore::ServiceWorkerRegistrationBackgroundFetchAPI::supplementName):
* Source/WebCore/workers/service/background-fetch/ServiceWorkerRegistrationBackgroundFetchAPI.h:
* Source/WebKit/NetworkProcess/Cookies/WebCookieManager.cpp:
(WebKit::WebCookieManager::supplementName):
* Source/WebKit/NetworkProcess/Cookies/WebCookieManager.h:
* Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolManager.cpp:
(WebKit::LegacyCustomProtocolManager::supplementName):
* Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolManager.h:
* Source/WebKit/NetworkProcess/cocoa/LaunchServicesDatabaseObserver.h:
* Source/WebKit/NetworkProcess/cocoa/LaunchServicesDatabaseObserver.mm:
(WebKit::LaunchServicesDatabaseObserver::supplementName):
* Source/WebKit/Shared/Authentication/AuthenticationManager.cpp:
(WebKit::AuthenticationManager::supplementName):
* Source/WebKit/Shared/Authentication/AuthenticationManager.h:
* Source/WebKit/UIProcess/Notifications/WebNotificationManagerProxy.cpp:
(WebKit::WebNotificationManagerProxy::supplementName):
* Source/WebKit/UIProcess/Notifications/WebNotificationManagerProxy.h:
* Source/WebKit/UIProcess/WebGeolocationManagerProxy.cpp:
(WebKit::WebGeolocationManagerProxy::supplementName):
* Source/WebKit/UIProcess/WebGeolocationManagerProxy.h:
* Source/WebKit/WebProcess/GPU/media/RemoteCDMFactory.cpp:
(WebKit::RemoteCDMFactory::supplementName):
* Source/WebKit/WebProcess/GPU/media/RemoteCDMFactory.h:
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMFactory.cpp:
(WebKit::RemoteLegacyCDMFactory::supplementName):
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMFactory.h:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaEngineConfigurationFactory.cpp:
(WebKit::RemoteMediaEngineConfigurationFactory::supplementName):
* Source/WebKit/WebProcess/GPU/media/RemoteMediaEngineConfigurationFactory.h:
* Source/WebKit/WebProcess/Geolocation/WebGeolocationManager.cpp:
(WebKit::WebGeolocationManager::supplementName):
* Source/WebKit/WebProcess/Geolocation/WebGeolocationManager.h:
* Source/WebKit/WebProcess/MediaCache/WebMediaKeyStorageManager.cpp:
(WebKit::WebMediaKeyStorageManager::supplementName):
* Source/WebKit/WebProcess/MediaCache/WebMediaKeyStorageManager.h:
* Source/WebKit/WebProcess/Notifications/WebNotificationManager.cpp:
(WebKit::WebNotificationManager::supplementName):
* Source/WebKit/WebProcess/Notifications/WebNotificationManager.h:
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/cocoa/AudioSessionRoutingArbitrator.cpp:
(WebKit::AudioSessionRoutingArbitrator::supplementName):
* Source/WebKit/WebProcess/cocoa/AudioSessionRoutingArbitrator.h:
* Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.cpp:
(WebKit::UserMediaCaptureManager::supplementName):
* Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.h:
* Source/WebKit/WebProcess/glib/UserMediaCaptureManager.h:
(WebKit::UserMediaCaptureManager::supplementName):

Canonical link: <a href="https://commits.webkit.org/272955@main">https://commits.webkit.org/272955@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a569410b851d90519b03929064f54e4fbf1b746d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/33721 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/12494 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/35656 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36336 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/30574 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/34769 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/14866 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/9647 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/29651 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34198 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/10535 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30063 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9193 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9286 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/30078 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/37659 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/28721 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/30603 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30397 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35417 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/33663 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/9428 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/7372 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/33305 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11208 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/40187 "Built successfully") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/10011 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/8411 "Passed tests") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4344 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10214 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WTF/wtf/text/ASCIILiteral.h:
```diff
@@ -122,6 +123,12 @@ struct ASCIILiteralHash {
 
+struct ASCIILiteralPtrHash {
+    static unsigned hash(const ASCIILiteral& key) { return IntHash<uintptr_t>::hash(reinterpret_cast<uintptr_t>(key.characters())); }
+    static bool equal(const ASCIILiteral& a, const ASCIILiteral& b) { return a.characters() == b.characters(); }
```
Is it guaranteed that the linker will unique all ASCII literal pointers? Feels like it _might_ be, but I'm not sure.

cdumez replied:
I don't think it does but this is not an issue here.

ASCIILiteral keeps using the other ASCIILiteralHash in general.

This new ASCIILiteralPtrHash is only for the SupplementMaps because they were already using PtrHash with `const char*`. Supplementable already required that keys (when they were of type `const char*`) were unique instances of a string literals.

Therefore, I carried over this behavior when switching over to ASCIILiteral to make sure I don't make perf worse.

geoffreygaren replied:
Maybe not guaranteed?

> Evaluating a [string-literal](https://timsong-cpp.github.io/cppwp/lex.string#nt:string-literal) results in a string literal object with static storage duration ([[basic.stc]](https://timsong-cpp.github.io/cppwp/basic.stc))[.](https://timsong-cpp.github.io/cppwp/lex.string#9.sentence-1) Whether all [string-literal](https://timsong-cpp.github.io/cppwp/lex.string#nt:string-literal)s are distinct (that is, are stored in nonoverlapping objects) and whether successive evaluations of a [string-literal](https://timsong-cpp.github.io/cppwp/lex.string#nt:string-literal) yield the same or a different object is unspecified[.](https://timsong-cpp.github.io/cppwp/lex.string#9.sentence-2)

https://timsong-cpp.github.io/cppwp/lex.string#9

cdumez replied:
I did not change behavior AFAIK. The supplementable maps were already using PtrHash with `const char*`. All supplementable keys come from static functions on their respective classes.

geoffreygaren reviewed (approved):
r=me

cdumez commented on Source/WebCore/platform/Supplementable.h:
```diff
@@ -121,7 +122,7 @@ class Supplementable {
 #endif
 
 private:
-    typedef HashMap<const char*, std::unique_ptr<Supplement<T>>, PtrHash<const char*>> SupplementMap;
```
Note how the previous code was using `PtrHash` here. I'm kept pre-existing behavior except that the const char* is now wrapped in an ASCIILiteral.

cdumez committed:
Use ASCIILiteral instead of `const char*` for Supplementable key
https://bugs.webkit.org/show_bug.cgi?id=267431

Reviewed by Geoffrey Garen.

It makes string lifetime clearer.

* Source/WTF/wtf/text/ASCIILiteral.h:
(WTF::ASCIILiteralPtrHash::hash):
(WTF::ASCIILiteralPtrHash::equal):
* Source/WebCore/Modules/async-clipboard/NavigatorClipboard.cpp:
(WebCore::NavigatorClipboard::supplementName):
* Source/WebCore/Modules/async-clipboard/NavigatorClipboard.h:
* Source/WebCore/Modules/audiosession/NavigatorAudioSession.cpp:
(WebCore::NavigatorAudioSession::supplementName):
* Source/WebCore/Modules/audiosession/NavigatorAudioSession.h:
* Source/WebCore/Modules/beacon/NavigatorBeacon.cpp:
(WebCore::NavigatorBeacon::supplementName):
* Source/WebCore/Modules/beacon/NavigatorBeacon.h:
* Source/WebCore/Modules/cache/WindowOrWorkerGlobalScopeCaches.cpp:
(WebCore::DOMWindowCaches::supplementName):
(WebCore::WorkerGlobalScopeCaches::supplementName):
* Source/WebCore/Modules/contact-picker/NavigatorContacts.cpp:
(WebCore::NavigatorContacts::supplementName):
* Source/WebCore/Modules/contact-picker/NavigatorContacts.h:
* Source/WebCore/Modules/cookie-consent/NavigatorCookieConsent.h:
* Source/WebCore/Modules/credentialmanagement/NavigatorCredentials.cpp:
(WebCore::NavigatorCredentials::supplementName):
* Source/WebCore/Modules/credentialmanagement/NavigatorCredentials.h:
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemController.cpp:
(WebCore::MediaKeySystemController::supplementName):
* Source/WebCore/Modules/encryptedmedia/MediaKeySystemController.h:
* Source/WebCore/Modules/gamepad/NavigatorGamepad.cpp:
(WebCore::NavigatorGamepad::supplementName):
* Source/WebCore/Modules/gamepad/NavigatorGamepad.h:
* Source/WebCore/Modules/geolocation/GeolocationController.cpp:
(WebCore::GeolocationController::supplementName):
* Source/WebCore/Modules/geolocation/GeolocationController.h:
* Source/WebCore/Modules/geolocation/NavigatorGeolocation.cpp:
(WebCore::NavigatorGeolocation::supplementName):
* Source/WebCore/Modules/geolocation/NavigatorGeolocation.h:
* Source/WebCore/Modules/indexeddb/WindowOrWorkerGlobalScopeIndexedDatabase.cpp:
(WebCore::DOMWindowIndexedDatabase::supplementName):
(WebCore::WorkerGlobalScopeIndexedDatabase::supplementName):
* Source/WebCore/Modules/mediacapabilities/NavigatorMediaCapabilities.cpp:
(WebCore::NavigatorMediaCapabilities::supplementName):
* Source/WebCore/Modules/mediacapabilities/NavigatorMediaCapabilities.h:
* Source/WebCore/Modules/mediacapabilities/WorkerNavigatorMediaCapabilities.cpp:
(WebCore::WorkerNavigatorMediaCapabilities::supplementName):
* Source/WebCore/Modules/mediacapabilities/WorkerNavigatorMediaCapabilities.h:
* Source/WebCore/Modules/mediasession/NavigatorMediaSession.cpp:
(WebCore::NavigatorMediaSession::supplementName):
* Source/WebCore/Modules/mediasession/NavigatorMediaSession.h:
* Source/WebCore/Modules/mediastream/NavigatorMediaDevices.cpp:
(WebCore::NavigatorMediaDevices::supplementName):
* Source/WebCore/Modules/mediastream/NavigatorMediaDevices.h:
* Source/WebCore/Modules/mediastream/UserMediaController.cpp:
(WebCore::UserMediaController::supplementName):
* Source/WebCore/Modules/mediastream/UserMediaController.h:
* Source/WebCore/Modules/notifications/NotificationController.cpp:
(WebCore::NotificationController::supplementName):
* Source/WebCore/Modules/notifications/NotificationController.h:
* Source/WebCore/Modules/permissions/NavigatorPermissions.cpp:
(WebCore::NavigatorPermissions::supplementName):
* Source/WebCore/Modules/permissions/NavigatorPermissions.h:
* Source/WebCore/Modules/permissions/WorkerNavigatorPermissions.cpp:
(WebCore::WorkerNavigatorPermissions::supplementName):
* Source/WebCore/Modules/permissions/WorkerNavigatorPermissions.h:
* Source/WebCore/Modules/pictureinpicture/DocumentPictureInPicture.h:
(WebCore::DocumentPictureInPicture::supplementName):
* Source/WebCore/Modules/pictureinpicture/HTMLVideoElementPictureInPicture.h:
(WebCore::HTMLVideoElementPictureInPicture::supplementName):
* Source/WebCore/Modules/push-api/ServiceWorkerRegistrationPushAPI.cpp:
(WebCore::ServiceWorkerRegistrationPushAPI::supplementName):
* Source/WebCore/Modules/push-api/ServiceWorkerRegistrationPushAPI.h:
* Source/WebCore/Modules/screen-wake-lock/NavigatorScreenWakeLock.cpp:
(WebCore::NavigatorScreenWakeLock::supplementName):
* Source/WebCore/Modules/screen-wake-lock/NavigatorScreenWakeLock.h:
* Source/WebCore/Modules/speech/LocalDOMWindowSpeechSynthesis.cpp:
(WebCore::LocalDOMWindowSpeechSynthesis::supplementName):
* Source/WebCore/Modules/speech/LocalDOMWindowSpeechSynthesis.h:
* Source/WebCore/Modules/webdriver/NavigatorWebDriver.cpp:
(WebCore::NavigatorWebDriver::supplementName):
* Source/WebCore/Modules/webdriver/NavigatorWebDriver.h:
* Source/WebCore/css/DOMCSSPaintWorklet.cpp:
(WebCore::DOMCSSPaintWorklet::supplementName):
* Source/WebCore/css/DOMCSSPaintWorklet.h:
* Source/WebCore/css/DOMCSSRegisterCustomProperty.cpp:
(WebCore::DOMCSSRegisterCustomProperty::supplementName):
* Source/WebCore/css/DOMCSSRegisterCustomProperty.h:
* Source/WebCore/css/typedom/CSSNumericFactory.cpp:
(WebCore::CSSNumericFactory::supplementName):
* Source/WebCore/css/typedom/CSSNumericFactory.h:
* Source/WebCore/dom/DeviceMotionController.cpp:
(WebCore::DeviceMotionController::supplementName):
* Source/WebCore/dom/DeviceMotionController.h:
* Source/WebCore/dom/DeviceOrientationController.cpp:
(WebCore::DeviceOrientationController::supplementName):
* Source/WebCore/dom/DeviceOrientationController.h:
* Source/WebCore/dom/DocumentStorageAccess.cpp:
(WebCore::DocumentStorageAccess::supplementName):
* Source/WebCore/dom/DocumentStorageAccess.h:
* Source/WebCore/html/NavigatorUserActivation.cpp:
(WebCore::NavigatorUserActivation::supplementName):
* Source/WebCore/html/NavigatorUserActivation.h:
* Source/WebCore/page/NavigatorIsLoggedIn.cpp:
(WebCore::NavigatorIsLoggedIn::supplementName):
* Source/WebCore/page/NavigatorIsLoggedIn.h:
* Source/WebCore/platform/Supplementable.h:
(WebCore::Supplement::provideTo):
(WebCore::Supplement::from):
(WebCore::Supplementable::provideSupplement):
(WebCore::Supplementable::removeSupplement):
(WebCore::Supplementable::requireSupplement):
* Source/WebCore/testing/InternalSettings.cpp:
(WebCore::InternalSettings::supplementName):
* Source/WebCore/testing/InternalSettings.h:
* Source/WebCore/workers/service/background-fetch/ServiceWorkerRegistrationBackgroundFetchAPI.cpp:
(WebCore::ServiceWorkerRegistrationBackgroundFetchAPI::supplementName):
* Source/WebCore/workers/service/background-fetch/ServiceWorkerRegistrationBackgroundFetchAPI.h:
* Source/WebKit/NetworkProcess/Cookies/WebCookieManager.cpp:
(WebKit::WebCookieManager::supplementName):
* Source/WebKit/NetworkProcess/Cookies/WebCookieManager.h:
* Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolManager.cpp:
(WebKit::LegacyCustomProtocolManager::supplementName):
* Source/WebKit/NetworkProcess/CustomProtocols/LegacyCustomProtocolManager.h:
* Source/WebKit/NetworkProcess/cocoa/LaunchServicesDatabaseObserver.h:
* Source/WebKit/NetworkProcess/cocoa/LaunchServicesDatabaseObserver.mm:
(WebKit::LaunchServicesDatabaseObserver::supplementName):
* Source/WebKit/Shared/Authentication/AuthenticationManager.cpp:
(WebKit::AuthenticationManager::supplementName):
* Source/WebKit/Shared/Authentication/AuthenticationManager.h:
* Source/WebKit/UIProcess/Notifications/WebNotificationManagerProxy.cpp:
(WebKit::WebNotificationManagerProxy::supplementName):
* Source/WebKit/UIProcess/Notifications/WebNotificationManagerProxy.h:
* Source/WebKit/UIProcess/WebGeolocationManagerProxy.cpp:
(WebKit::WebGeolocationManagerProxy::supplementName):
* Source/WebKit/UIProcess/WebGeolocationManagerProxy.h:
* Source/WebKit/WebProcess/GPU/media/RemoteCDMFactory.cpp:
(WebKit::RemoteCDMFactory::supplementName):
* Source/WebKit/WebProcess/GPU/media/RemoteCDMFactory.h:
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMFactory.cpp:
(WebKit::RemoteLegacyCDMFactory::supplementName):
* Source/WebKit/WebProcess/GPU/media/RemoteLegacyCDMFactory.h:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaEngineConfigurationFactory.cpp:
(WebKit::RemoteMediaEngineConfigurationFactory::supplementName):
* Source/WebKit/WebProcess/GPU/media/RemoteMediaEngineConfigurationFactory.h:
* Source/WebKit/WebProcess/Geolocation/WebGeolocationManager.cpp:
(WebKit::WebGeolocationManager::supplementName):
* Source/WebKit/WebProcess/Geolocation/WebGeolocationManager.h:
* Source/WebKit/WebProcess/MediaCache/WebMediaKeyStorageManager.cpp:
(WebKit::WebMediaKeyStorageManager::supplementName):
* Source/WebKit/WebProcess/MediaCache/WebMediaKeyStorageManager.h:
* Source/WebKit/WebProcess/Notifications/WebNotificationManager.cpp:
(WebKit::WebNotificationManager::supplementName):
* Source/WebKit/WebProcess/Notifications/WebNotificationManager.h:
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKit/WebProcess/cocoa/AudioSessionRoutingArbitrator.cpp:
(WebKit::AudioSessionRoutingArbitrator::supplementName):
* Source/WebKit/WebProcess/cocoa/AudioSessionRoutingArbitrator.h:
* Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.cpp:
(WebKit::UserMediaCaptureManager::supplementName):
* Source/WebKit/WebProcess/cocoa/UserMediaCaptureManager.h:
* Source/WebKit/WebProcess/glib/UserMediaCaptureManager.h:
(WebKit::UserMediaCaptureManager::supplementName):

Canonical link: https://commits.webkit.org/272955@main
